src/amo/containers/App.js

```diff
@@ -57,10 +59,10 @@ export class AppBase extends React.Component {
   }
 
   accountButton() {
-    const { handleLogIn, i18n, isAuthenticated, location } = this.props;
+    const { handleLogIn, handleLogOut, i18n, isAuthenticated, location } = this.props;
     return (
       <button className="button AccountButton"
-              onClick={() => handleLogIn(location)}
+              onClick={() => (isAuthenticated ? handleLogOut() : handleLogIn(location))}
```

Comment:
This is a good call. I did however end up moving the button into its own component and that is now included by `MastHead` instead of being the only passed in child to it from `App`. I also refactored the CSS in `MastHead` a bit for the header to use flexbox  (I added a `div` but it's shorter) and an `Icon` (in `AuthenticateButton`).